sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -557,6 +557,348 @@ inline __SYCL_ALWAYS_INLINE void joint_matrix_apply(
+        spv_matrix_layout_traits<layout::dynamic>::value>(
+        Ptr, stride, Height, Width, __spv::MatrixLayout::ColumnMajor,
+        spv_scope_traits<Group>::value);
+    break;
+  case layout::ext_intel_packed:
+    res.spvm = __spirv_JointMatrixLoadCheckedINTEL<
+        T, S, NumRows, NumCols, spv_matrix_use_traits<use::accumulator>::value,
+        spv_matrix_layout_traits<layout::dynamic>::value>(
+        Ptr, stride, Height, Width, __spv::MatrixLayout::Packed,
+        spv_scope_traits<Group>::value);
+    break;
+  }
+#else
+  std::ignore = sg;
+  std::ignore = res;
+  std::ignore = src;
+  std::ignore = stride;
+  std::ignore = Layout;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <
+    typename Group, typename S, typename T, use Use, size_t NumRows,
+    size_t NumCols, layout Layout, typename PropertyListT,
+    std::enable_if_t<std::is_same<S, std::remove_const_t<T>>::value ||
+                         (std::is_same<S, precision::tf32>::value &&
+                          std::is_same<std::remove_const_t<T>, float>::value),
+                     bool> = true>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_load_checked(
+    Group sg, joint_matrix<Group, S, Use, NumRows, NumCols, Layout> &res,
+    ext::oneapi::experimental::annotated_ptr<T, PropertyListT> src,
+    size_t stride, size_t Height, size_t Width) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  T *Ptr = src.get();
+  res.spvm = __spirv_JointMatrixLoadCheckedINTEL<
+      T, S, NumRows, NumCols, spv_matrix_use_traits<Use>::value,
+      spv_matrix_layout_traits<Layout>::value>(
+      Ptr, stride, Height, Width, spv_matrix_layout_traits<Layout>::value,
+      spv_scope_traits<Group>::value);
+#else
+  std::ignore = sg;
+  std::ignore = res;
+  std::ignore = src;
+  std::ignore = stride;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          typename PropertyListT>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_store_checked(
+    Group sg,
+    joint_matrix<Group, T, use::accumulator, NumRows, NumCols, layout::dynamic>
+        &src,
+    ext::oneapi::experimental::annotated_ptr<T, PropertyListT> dst,
+    size_t stride, layout Layout, size_t Height, size_t Width) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  T *Ptr = dst.get();
+  switch (Layout) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case layout::row_major:
+    __spirv_JointMatrixStoreCheckedINTEL<
+        T, T, NumRows, NumCols, spv_matrix_use_traits<use::accumulator>::value,
+        spv_matrix_layout_traits<layout::dynamic>::value>(
+        Ptr, src.spvm, stride, Height, Width, __spv::MatrixLayout::RowMajor,
```

Comment:
Same
```suggestion
        Ptr, src.spvm, stride, Height, Width, spv_matrix_layout_traits<Layout>::value,
```

Reply:
No, it is not possible because layout in this case is a runtime variable.

Reply:
https://github.com/intel/llvm/pull/11210#discussion_r1431849828